UKHomeOffice/nmsw-ui
# NMSW-424 Show passengers as completed if declaration has passengers

# Ticket

NMSW-424

----

## AC

If GET declaration endpoint results.passengers === undefined then passengers is NOT completed

If GET declaration endpoint results.passengers === NO then passengers IS completed

If GET declaration endpoint results.passengers === YES 
AND GET passengers file upload === undefined then passengers is NOT completed

If GET declaration endpoint results.passengers === YES
AND GET passengers file upload === a file then passengers IS completed

----

## To test

- Run app
- Sign in
- Create a report
- Complete General declaration FAL 1
- > On the tasklist page passengers should be required
- Click on Any passenger details
- Select yes to passengers
- On the file upload page click back until you get back to the tasklist page
- > The passenger details should still say required
- Click on Any passenger details
- Select yes
- Upload a good file
- > On the task details page the passenger details should be completed
- Create a new report
- Complete General declaration FAL 1
- On the tasklist page click on Any passenger details
- Select no to passengers
- On the task details page the passenger details should be completed

----

## Notes

- If a user selects 'yes' to passengers then uploads a file, if they then click 'no' to passengers, the file will still exist
